libs/dh/market-participant/actors/feature-delegation/src/lib/dh-delegation-tab.component.ts

```diff
@@ -101,15 +107,23 @@ export class DhDelegationTabComponent {
   statusOptions = dhEnumToWattDropdownOptions(ActorDelegationStatus);
 
   constructor() {
-    effect(() => this.fetchData(this.actor().id), { allowSignalWrites: true });
+    effect(() => {
+      this.statusControl.reset();
+
+      this.delegationsForActorQuery.refetch({ actorId: this.actor().id });
+    });
 
-    this.statusControl.valueChanges.subscribe((value) => {
-      this.filterDelegations(value);
+    effect(() => this.delegationsByType.set(dhGroupDelegations(this.delegationsRaw())), {
+      allowSignalWrites: true,
     });
+
+    this.statusControl.valueChanges
+      .pipe(takeUntilDestroyed(this.destroyRef))
```

Comment:
Suggestion: No need for a destoryRef inside a constructor

Reply:
Fixed